### DISCUSSION TITLE
Grapqhl and relay returns module for typescript

### DISCUSSION THREAD

**Discussion by EloB** [vercel/next.js#36365] [Help]
I've been trying to implement relay and typescript. My inspiration came from the latest data-driven-dependencies example that is written in javascript.
https://github.com/relayjs/relay-examples/blob/main/data-driven-dependencies/pages/index.js#L8

My problem in my codebase is that:
```ts
const someNameQuery = graphql`
  query someNameQuery @preloadable {
    hello { world }
  }
`;

console.log(someNameQuery); // outputs: Object [Module] { default: [Getter] }
```

What I expect it to output is:
```ts
{
  fragment: {
    argumentDefinitions: [],
    kind: 'Fragment',
    metadata: null,
    name: 'someNameQuery',
    selections: [ [Object] ],
    type: 'query_root',
    abstractKey: null
  },
  kind: 'Request',
  operation: {
    argumentDefinitions: [],
    kind: 'Operation',
    name: 'someNameQuery',
    selections: [ [Object] ]
  },
  params: {
    id: 'dcf40f930f6040061bd3f7f57362fdd2760b2153',
    metadata: {},
    name: 'someNameQuery',
    operationKind: 'query',
    text: null
  },
  hash: '8987d6c85ebe750ce47d16076e5f1cae'
}
```

I'm been trying to change `tsconfig.json` but haven't got it to work and I'm not even sure that is the problem...

**Comment by EloB** [comment]
Here is a example:
https://codesandbox.io/s/misty-surf-1vh56k?file=/pages/index.tsx

**Comment by icyJoseph** [comment]
I looked a bit into the react-relay source code and played around with the codesandbox, and I must say that this ought to be a problem with https://github.com/facebook/relay/tree/main/packages/babel-plugin-relay, I'd also ask this question again over there.

**Reply by EloB** [reply]
@icyJoseph How is that babel plugin used? Isn't SWC supporting relay out of the box now or does SWC somehow use babel plugins? :)

**Reply by icyJoseph** [reply]
Yeah, right, like I said, I let the source code guide my hand on this one. 

I still wonder though, if the babel implementation outputs a plain object, why is the SWC implementation outputting a module. I guess, then, someone who's been following the SWC development should help out here.

I found this thread at the relay repo where you could also forward this question: https://github.com/facebook/relay/issues/3319

**Reply by EloB** [reply]
Thanks for your effort and digging! 🙂

**Reply by EloB** [reply]
There is a issue right now at SWC atleast.
https://github.com/swc-project/plugins/issues/107
